File: moveit/moveit_action_handlers/python_test_clients/move_joints_gripper_test_client.py
#! /usr/bin/env python
import rospy
# Brings in the SimpleActionClient
import actionlib

from moveit_action_handlers.msg import MoveToJointsMoveItAction
from moveit_action_handlers.msg import MoveToJointsMoveItGoal
# property value pair joint names joint values
from moveit_action_handlers.msg import PropertyValuePair


def moveit_tojoints_client():
    # ARM
    gripper_client = actionlib
    gripper_client = actionlib.simple_action_client.SimpleActionClient(
        'move_tojoints_action_server/gripper/action', MoveToJointsMoveItAction)

    gripper_client.wait_for_server()

    # Creates a goal to send to the action server gripper.

    goal = MoveToJointsMoveItGoal()
    goal.endEffectorVelocity = 0.5
    goal.endEffectorAcceleration = 0.5
    goal.timeoutSeconds = 10
    # gripper joint names list
    joint_names = ["finger1_joint", "finger2_joint",]
    # gripper joint values close
    joint_values = [0.02, 0.02]
    # gripper joint values open
    #joint_values = [0.0, 0.0]
    
    # create and fill joint value pairs to goal
    for i in range(len(joint_names)):
        joint_pair = PropertyValuePair()
        joint_pair.name = joint_names[i]
        joint_pair.value = joint_values[i]
        # append to goal
        goal.joint_pairs.append(joint_pair)
    


    # send goal
    gripper_client.send_goal(goal)

    # Waits for the server to finish performing the action.
    gripper_client.wait_for_result()


if __name__ == '__main__':

    rospy.init_node('test_joints_client_python_node')
    result = moveit_tojoints_client()

File: moveit/moveit_action_handlers/cpp_test_clients/move_joints_gripper_test_client.cpp

#include <actionlib/client/simple_action_client.h>
#include <actionlib/client/terminal_state.h>
#include <moveit_action_handlers/MoveToJointsMoveItAction.h>
#include <moveit_action_handlers/MoveToJointsMoveItGoal.h>
#include <moveit_action_handlers/PropertyValuePair.h>
#include <ros/ros.h>
#include <string>

int main(int argc, char **argv) {
  ros::init(argc, argv, "test_joints_gripper_client_cpp_node");

  // create the action client
  // true causes the client to spin its own thread
  actionlib::SimpleActionClient<moveit_action_handlers::MoveToJointsMoveItAction> action_client_gripper(
      "move_tojoints_action_server/gripper/action", true);

  ROS_INFO("Waiting for action server to start.");
  // wait for the action server to start
  action_client_gripper.waitForServer(); // will wait for infinite time

  // Creates a goal to send to the action server arm.
  moveit_action_handlers::MoveToJointsMoveItGoal goal;
  goal.endEffectorVelocity = 0.5;
  goal.endEffectorAcceleration = 0.5;
  goal.timeoutSeconds = 20;
  //  joint names
  std::vector<std::string> joint_names = {"finger1_joint", "finger2_joint"};

  // gripper joint values close
  std::vector<float> joint_values = {0.02, 0.02};
  // gripper joint values open
  // std::vector<float> joint_values = {0.0, 0.0};
  for (size_t i = 0; i < joint_names.size(); i++) {
    moveit_action_handlers::PropertyValuePair joint_pair;
    joint_pair.name = joint_names[i];
    joint_pair.value = joint_values[i];
    // push back to goal the joint pair
    goal.joint_pairs.push_back(joint_pair);
  }

  // send goal
  action_client_gripper.sendGoal(goal);

  // wait for the action to return
  bool finished_before_timeout = action_client_gripper.waitForResult(ros::Duration(20.0));

  if (finished_before_timeout) {
    actionlib::SimpleClientGoalState state = action_client_gripper.getState();
    ROS_INFO("Action finished: %s", state.toString().c_str());
  } else
    ROS_INFO("Action did not finish before the time out.");

  // exit
  return 0;
}